internal/ent/schema/apitoken.go

```diff
@@ -63,6 +63,20 @@ func (APIToken) Fields() []ent.Field {
 				token := keygen.PrefixedSecret("tola") // api token prefix
 				return token
 			}),
+		field.String("token_public_id").
+			Unique().
+			Immutable().
+			Optional().
+			Nillable().
+			Comment("public identifier for the token (used to lookup token record)"),
+		field.String("token_secret").
+			Immutable().
+			Optional().
+			Nillable().
```

Comment:
This needs to be `Unique()`
```suggestion
			Optional().
			Unique().
			NotEmpty().
```